src/main/java/me/rigamortis/seppuku/impl/module/combat/NoCrystalModule.java

```diff
@@ -133,37 +124,53 @@ private int findStackHotbar(Block type) {
             final ItemStack stack = Minecraft.getMinecraft().player.inventory.getStackInSlot(i);
             if (stack.getItem() instanceof ItemBlock) {
                 final ItemBlock block = (ItemBlock) stack.getItem();
-
-                if (block.getBlock() == type) {
-                    return i;
-                }
+                if (block.getBlock() == type) return i;
             }
         }
         return -1;
     }
 
     private boolean valid (BlockPos pos) {
         // There are no entities to block placement,
-        if (!mc.world.checkNoEntityCollision(new AxisAlignedBB(pos))) {
-            return false;
-        }
-
+        if (!mc.world.checkNoEntityCollision(new AxisAlignedBB(pos))) return false;
         // Check if the block is replaceable
         return mc.world.getBlockState(pos).getBlock().isReplaceable(mc.world, pos);
     }
 
-    private void place (BlockPos pos, EnumFacing direction) {
+    private void place(BlockPos pos) {
         final Block block = mc.world.getBlockState(pos).getBlock();
+        final EnumFacing direction = calcSide(pos);
         final boolean activated = block.onBlockActivated(mc.world, pos, mc.world.getBlockState(pos), mc.player, EnumHand.MAIN_HAND, direction, 0, 0, 0);
 
-        if (activated) {
+        if (activated)
             mc.player.connection.sendPacket(new CPacketEntityAction(mc.player, CPacketEntityAction.Action.START_SNEAKING));
-        }
-
-        this.mc.player.connection.sendPacket(new CPacketPlayerTryUseItemOnBlock(pos, direction, EnumHand.MAIN_HAND, 0.0F, 0.0F, 0.0F));
+        if (direction != null) {
+            final EnumFacing otherSide = direction.getOpposite();
+            final BlockPos sideOffset = pos.offset(direction);
 
-        if (activated) {
+            if (rotate.getValue()) {
+                final float[] angle = MathUtil.calcAngle(mc.player.getPositionEyes(mc.getRenderPartialTicks()), new Vec3d(pos.getX() + 0.5f, pos.getY() + 0.5f, pos.getZ() + 0.5f));
+                Seppuku.INSTANCE.getRotationManager().setPlayerRotations(angle[0], angle[1]);
+            }
+            if (mode.getValue() == Mode.PACKET) {
+                mc.player.connection.sendPacket(new CPacketPlayerTryUseItemOnBlock(sideOffset, otherSide, EnumHand.MAIN_HAND, 0.5F, 0.5F, 0.5F));
+                mc.player.swingArm(EnumHand.MAIN_HAND);
```

Comment:
Since this is a packet based mode, why not make the swing based on a packet as well?

Reply:
Whoops, forgot to change this. Fixed.

Reply:
I had forgotten to do this completely. Thanks for the reminder.